controllers/userController.js

```diff
@@ -62,6 +64,69 @@ const addUser = async (req, res) => {
   }
 };
 
+/**
+ * Controller for adding users from csv file
+ * @param {Request} req Request Object
+ * @param {Response} res Response Object
+ */
+const addUsersFromFile = async function (req, res) {
+  const file = req.file;
+  let str = undefined;
+  let users = [];
+  try {
+    str = readFileSync(file.path).toString();
+    const { data } = parse(str);
+    users = data.slice(1, -1);
+  } catch (error) {
+    return res
+      .status(500)
+      .send(new Error("INTERNAL_SERVER_ERROR", error.message));
+  } finally {
+    unlinkSync(file.path);
+  }
+
+  // Array to store zprn of users not created due to any error
+  const erroneousUsers = [];
+
+  users.forEach(async (user) => {
+    const [zprn, firstName, lastName, password, department, role] = user;
+    const { error } = validateData(
+      zprn,
+      firstName,
+      lastName,
+      password,
+      department,
+      role
+    );
+
+    if (error) {
+      erroneousUsers.push(zprn);
```

Comment:
I think we shouldn't add any user if there is an error with some user. Because if there are errors we should just return the `erroneous` users list and let the end user make changes in the csv regarding the same and then reupload the whole csv. Because some end users might reupload the same csv multiple times, creating more errors because of same zprn again repeating.

Reply:
Okay.

Reply:
Also lets make that clear in documentation and even in the response message that after there is an error, no user is added.